torchbenchmark/e2e_models/hf_t5/__init__.py

```diff
@@ -0,0 +1,389 @@
+from accelerate.utils.dataclasses import DeepSpeedPlugin
+import torch
+import numpy as np
+import math
+import os
+from pathlib import Path
+from torch.nn.parallel import DistributedDataParallel as DDP
+from torch.utils.data import DataLoader
+from torchbenchmark.util.e2emodel import E2EBenchmarkModel
+from torchbenchmark.tasks import NLP
+
+import evaluate
+from accelerate import Accelerator
+from transformers import (
+    CONFIG_MAPPING,
+    AutoConfig,
+    AutoModelForSeq2SeqLM,
+    AutoTokenizer,
+    DataCollatorForSeq2Seq,
+    default_data_collator,
+    get_scheduler,
+    MBartTokenizer,
+    MBartTokenizerFast
+)
+from torchbenchmark.util.framework.transformers.translation.dataset import prep_dataset, preprocess_dataset
+from torchbenchmark.util.framework.transformers.translation.args import parse_args, parse_torchbench_args, task_to_keys
+
+# setup environment variable
+CURRENT_DIR = Path(os.path.dirname(os.path.realpath(__file__)))
+
+class Model(E2EBenchmarkModel):
+    task = NLP.TRANSLATION
+    DEFAULT_TRAIN_BSIZE: int = 32
+    DEFAULT_EVAL_BSIZE: int = 1
+
+    def __init__(self, test, batch_size=None, extra_args=[]):
+        super().__init__(test=test, batch_size=batch_size, extra_args=extra_args)
+        self.device = "cuda"
+        self.device_num = 1
+        # Parse the extra arguments
+        self.tb_args = parse_torchbench_args(extra_args)
+        torch.manual_seed(1337)
+        torch.backends.cudnn.deterministic = False
+        torch.backends.cudnn.benchmark = True
+
+        # Parameters
+        model_name = "t5-base"
+        max_source_length = "1024"
+        max_target_length = "128"
+        learning_rate = "2e-5"
+        num_train_epochs = "3" # this takes a rather long time for wmt-en-ro
+        max_train_steps = "1000" # overrides num_train_epochs to run faster
+        checkpointing_steps = None # set to a string value, like "1000"
+
+        task_name = self.tb_args.task_name
+        task_args = task_to_keys[task_name] # dataset specific hf_args
+        # T5 requires source prefix to know what to translate
+        if task_name == "wmt-en-ro":
+            source_prefix = "translate English to Romanian: "
+        elif task_name == "wmt-en-de":
+            source_prefix = "translate English to German: "
+        else:
+            raise RuntimeError(f"Unsupported translation task {task_name} for model hf_t5")
+        task_args.extend(["--source_prefix", source_prefix])
```

Comment:
Counterpart to [comment](https://github.com/pytorch/benchmark/pull/1122/files#r955279905) below - is this the best abstraction to separate translation task specific args and model specific args?

Reply:
Yes, I think it is a good abstraction.